### PR TITLE
Add alternative trimmer: AdapterRemoval

### LINKED ISSUE #284: Add AdapterRemoval as alternative clipping tool
## Is your feature request related to a problem? Please describe

AdapterRemoval is commonly used in aDNA, so it would be a useful addition to sync with other 'common' workflows for that.

## Describe the solution you'd like

Add AR2 as an alternative trimmer

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Fixes #284 

Will require https://github.com/nf-core/test-datasets/pull/482 to be merged before `test_adapterremoval` works

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: docs/output.md [hunk 1]
```diff
@@ -57,6 +57,20 @@ FastQC is run for visualising the general quality metrics of the sequencing runs
 
 </details>
 
+### AdapterRemoval2
+
+[fastp](https://github.com/OpenGene/fastp) is a all-in-one fastq preprocessor for read/adapter trimming and quality control. It is used in this pipeline for trimming adapter sequences and discard low-quality reads. Its output is in the results folder and part of the MultiQC report.
```
Comment (d4straub) [inline docs/output.md:62]:
```suggestion
```

File: workflows/mag.nf [hunk 2]
```diff
@@ -223,13 +225,45 @@ workflow MAG {
...
+
+        ADAPTERREMOVAL_PE ( ch_adapterremoval_in.paired, [] )
+        ADAPTERREMOVAL_SE ( ch_adapterremoval_in.single, [] )
+
+        // pair1 and 2 come for PE data from separate output channels, so bring
+        // this back together again here
+        ch_adapterremoval_pe_out = Channel.empty()
+        ch_adapterremoval_pe_out = ADAPTERREMOVAL_PE.out.pair1_truncated
+            .join(ADAPTERREMOVAL_PE.out.pair2_truncated)
+            .dump(tag: "ar_pe_mix_out")
```
Comment (d4straub) [inline workflows/mag.nf:255]: If I didnt miss anything, you typically remove the `.dump()` usually?
```suggestion
```

File: workflows/mag.nf [hunk 3]
```diff
@@ -223,13 +225,45 @@ workflow MAG {
...
+            .map {
+                [ it[0], [it[1], it[2]] ]
+            }
+            .dump(tag: "ar2_pe_out")
```
Comment (d4straub) [inline workflows/mag.nf:259]:
```suggestion
```

**Review by d4straub — COMMENTED** [review]
Looks all good apart from a few cosmetic changes and the not running tests (the latter is important I guess).

File: workflows/mag.nf (hunk 2 above)
Comment (jfy133) [reply workflows/mag.nf:255]: Yes correct, normally remove as it seems you guys don't leave it in.

**Review by d4straub — APPROVED** [review]
LGTM!

**Pull request merged**